NebulaWorld/Factory/PowerTowerManager.cs

```diff
@@ -225,7 +224,11 @@ public static void AddExtraDemand(int PlanetId, int NetId, int NodeId, int Power
                 map.ExtraPower = PowerAmount;
 
                 mapping2.Add(map);
-                Energy.Add(PlanetId, mapping2);
+                if (!Energy.TryAdd(PlanetId, mapping2))
+                {
+                    // if we failed to add then most likely because another thread was faster, so call this again to run the above part of the method.
```

Comment:
Is it possible for this to keep failing indefinitely? Should there be a condition where it stops trying?

Reply:
I dont think so, this only happens if some player walks into the range of a power tower, and then it calls it once per tower.
I think this is only relevant anyways if you walk into the range for more than one tower at exactly the same time